_posts/2018-04-17-smartroom-api.md

```diff
@@ -0,0 +1,45 @@
+---
+layout: post
+title: Smartroom API
+date: 2018-04-17
+categories:
+    - Project
+    - PHP
+    - Lumen
+    - REST
+    - API
+description: A uniffed URL for accessing the varius parts of CSH's Smart Room Display
+image: https://fthmb.tqn.com/FzECgdCy_W-gXosZw89_-8UDraw=/768x0/filters:no_upscale():max_bytes(150000):strip_icc():format(webp)/PHP-code-58d2d5803df78c51623a6ce2.jpg
+image-sm: https://fthmb.tqn.com/FzECgdCy_W-gXosZw89_-8UDraw=/768x0/filters:no_upscale():max_bytes(150000):strip_icc():format(webp)/PHP-code-58d2d5803df78c51623a6ce2.jpg
+author: Leul Berhane-Meskel
+author-image: 
+author-bio: A 3rd year Computer Science student and an active member of CSH. A Developing Developer.
+author-email: leul@csh.rit.edu
+author-social:
+    github: github.com/LeulBM
+    linkedin: 
+---
+# One API to Rule them All
+In March, the idea of CSH creating an exhibit for ImagineRIT 2018 began floating around. We quickly decided on a Smart Dorm room, a glimps of what could be. It would be a great collaborative
+ project, with a multitude of unique parts creatied by various members, each person being allowed to work and create how they pleased. The only requirement is that the projects are internet 
+connected. The Unified API exists as a way to unify the development experience of user facing point of interaction with the Smart Room. Instead of being forced to consider several endpoint 
+naming schemes and data input formatting standards, the unified API allows developers to focus on User interactions rather than figuring out the back end minusha.
+
+# The Tech
+This API is written in PHP, utilizing the Lumen framework, specialized for microservices. The making of requests to individual aspects of the room  is handled by a Guzzle 6.0, a PHP package 
+made for crafting, sending, and parsing HTTP request and response objects. The source code for the API can be found in my [SmartRoomAPI Repo](https://github.com/LeulBM/SmartRoomAPI). 
+Check out the repo for detailed listing of all endpoints as well as links to the related repositories. The service is hosted using CSH's OpenShift environment at 
+[smartroom-api.csh.rit.edu](https://smartroom-api.csh.rit.edu).
+
+# But What Can It Do?
+Let's go down an example of how the API looks to a developer. Let's say you want to open the shades to the room. First, you need to know the URL you need to access. The naming scheme for the 
```

Comment:
Don't use "let's" to start two sentences in a row; you can start the second one with just "Say you..."